packages/nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -342,6 +337,41 @@ export class RedirectsMiddleware extends MiddlewareBase {
     return url;
   }
 
+  /**
+   * Helper function to dispatch a redirect or rewrite based on the redirect type.
+   * @param {NextURL | string} target The final target to redirect/rewrite to.
+   * @param {string} type One of `REDIRECT_TYPE_301`, `REDIRECT_TYPE_302`, or `REDIRECT_TYPE_SERVER_TRANSFER`.
+   * @param {NextRequest} req The incoming request.
+   * @param {NextResponse} res The current response (used for header cleanup / carry-over).
+   * @param {boolean} [isExternal=false] Whether `target` is an external absolute URL.
+   *   Passed through to `rewrite` so it can skip locale/basePath stripping for externals.
+   * @returns {NextResponse} The redirect/rewrite response, or `res` if the type is not recognized.
+   */
+  protected dispatchRedirect(
+    target: NextURL | string,
+    type: string,
+    req: NextRequest,
+    res: NextResponse,
+    isExternal = false
+  ): NextResponse {
+    switch (type) {
+      case REDIRECT_TYPE_301:
+        return this.createRedirectResponse(target, res, 301, 'Moved Permanently');
+      case REDIRECT_TYPE_302:
+        return this.createRedirectResponse(target, res, 302, 'Found');
+      case REDIRECT_TYPE_SERVER_TRANSFER:
+        // rewrite expects a string; unwrap NextURL if needed
+        return this.rewrite(
+          typeof target === 'string' ? target : target.href,
+          req,
+          res,
+          isExternal
+        );
+      default:
+        return res;
+    }
+  }
+
   /**
    * Helper function to create a redirect response and remove the x-middleware-next header.
    * @param {NextURL} url The URL to redirect to.
```

Comment:
The parameter's type should be updated
```suggestion
   * @param {NextURL | string} url The URL to redirect to.
```